File: Top 100 Codes/02-Working with Numbers/24-Finding Number of times x digit occurs in a given input/FindingNumberoftimesxdigitoccursinagiveninput.py
# Take user input for the number and digit
number_input = input("Enter a number: ")
digit_input = input("Enter a digit: ")

# Convert inputs to strings (they are already strings from input())
# but this step makes it clear we're working with string data
number_str = str(number_input)
digit_str = str(digit_input)

# Count the occurrences of the digit in the number
occurrence = number_str.count(digit_str)

# Print the result
print(f"The digit {digit_str} occurs {occurrence} time(s) in the number {number_str}.")

File: Top 100 Codes/02-Working with Numbers/24-Finding Number of times x digit occurs in a given input/FindingNumberoftimesxdigitoccursinagiveninput.cpp
#include <iostream>
using namespace std;

int main() {
    long long number;
    int digit;
    
    // Input the number and the digit to be counted
    cout << "Enter a number: ";
    cin >> number;
    cout << "Enter the digit to count: ";
    cin >> digit;
    
    int count = 0;
    
    // Process each digit of the number
    while (number != 0) {
        int remainder = number % 10;  // get the last digit
        
        if (remainder == digit) {
            count++;  // increment count if the digit matches
        }
        
        number /= 10;  // remove the last digit
    }
    
    cout << "The digit " << digit << " occurs " << count << " time(s) in the number." << endl;
    
    return 0;
}
